app/assets/javascripts/modules/sticky-element-container.js

```diff
@@ -0,0 +1,136 @@
+/*
+  This module will cause a child in the target element to:
+  - hide when the top of the target element is visible;
+  - stick to the bottom of the window while the parent element is in view;
+  - stick to the bottom of the target when the user scrolls past the bottom.
+
+  Use 'data-module="sticky-element-container"' to instantiate, and add
+  `[data-sticky-element]` to the child you want to position.
+*/
+(function (Modules, root) {
+  'use strict';
+
+  var $ = root.$;
+  var $window = $(root);
+
+  Modules.StickyElementContainer = function () {
+    var self = this;
+
+    self._getWindowDimensions = function _getWindowDimensions () {
+      return {
+        height: $window.height(),
+        width: $window.width()
+      };
+    };
+
+    self._getWindowPositions = function _getWindowPositions () {
+      return {
+        scrollTop: $window.scrollTop()
+      };
+    };
+
+    self.start = function start ($el) {
+      var $element = $el.find('[data-sticky-element]');
+      var _hasResized = true;
+      var _hasScrolled = true;
+      var _interval = 50;
+      var _isDesktop = true;
+      var _windowVerticalPosition = 1;
+      var _startPosition, _stopPosition;
+
+      initialise();
+
+      function initialise () {
+        $window.resize(onResize);
+        $window.scroll(onScroll);
+        setInterval(checkResize, _interval);
```

Comment:
Why `setInterval` rather than a `setTimeout` that sets itself to run again once it's reached the end?

If `checkResize` or `checkScroll` regularly takes longer than 50ms to run then the CPU might spike and the UI become less responsive (consider a low CPU mobile device). Similarly, if any of the code in those functions errors, the error would fire every 50ms, rather than stopping after the first error. For more details: http://stackoverflow.com/questions/729921/settimeout-or-setinterval